src/migrationTypes/changeColumnDefault.ts

```diff
@@ -16,30 +16,67 @@ export type ChangeColumnDefaultOptions = {
   /** The old default value */
   oldDefault: string;
   /** The new default value */
-  newDefault: string;
+  newDefault: string | null;
 };
 
 /**
- * Add a new version of the api
+ * Set the default value for a given column
  *
  * @param tableName - The table to change default for
  * @param columnName - The column to change default for
  * @param defaultValue - The default value to change to
  * @param transactionOptions - The transaction options
- * @param db - The database to operate on
  */
-export async function changeDefaultValue<TModels extends ModelMap>(
+export async function setColumnDefault<TModels extends ModelMap>(
   tableName: string,
   columnName: string,
   defaultValue: string,
   transactionOptions: MigrationTransactionOptions<TModels>,
 ): Promise<void> {
-  // Add the enum values to the column
   await transactionOptions.queryT.raw(
     `ALTER TABLE "${tableName}" ALTER "${columnName}" SET DEFAULT '${defaultValue}';`,
   );
 }
 
+/**
+ * Drops the default value for a given column
+ *
+ * @param tableName - The table to change default for
+ * @param columnName - The column to change default for
+ * @param defaultValue - The default value to change to
+ * @param transactionOptions - The transaction options
+ */
+export async function dropColumnDefault<TModels extends ModelMap>(
+  tableName: string,
+  columnName: string,
+  transactionOptions: MigrationTransactionOptions<TModels>,
+): Promise<void> {
+  await transactionOptions.queryT.raw(
+    `ALTER TABLE "${tableName}" ALTER "${columnName}" DROP DEFAULT;`,
+  );
+}
+
+/**
+ * Changes or drops the default value for a given column
+ *
+ * @param tableName - The table to change default for
+ * @param columnName - The column to change default for
+ * @param defaultValue - The default value to change to
+ * @param transactionOptions - The transaction options
+ */
+export async function changeDefaultValue<TModels extends ModelMap>(
+  tableName: string,
+  columnName: string,
+  defaultValue: string | null,
+  transactionOptions: MigrationTransactionOptions<TModels>,
+): Promise<void> {
+  if (defaultValue === null) {
+    dropColumnDefault(tableName, columnName, transactionOptions);
```

Comment:
be sure to await dropColumnDefault or setColumnDefault. this will return before the promise is actually resolved and will result in some weird race conditions

Reply:
Curious, is it not enough to await within the functions `dropColumnDefault` or `setColumnDefault` ?

Reply:
No it doesn't matter if the function calls await or not. The thing is that if the function returns a promise, whether it awaits internally or not, the caller of that function must await the promise to get the expected use case.

The following example should highlight the possible issue:

```ts
function add(x, y) {
 console.log('2')
 return x + y
}

/**
 * Sleep in a promise
 */
function sleepPromise(sleepTime) {
  return new Promise((resolve) =>
    setTimeout(() => resolve(sleepTime), sleepTime),
  );
}


async function addAsync(x, y) {
  await sleepPromise(100)
 console.log('4')
 return x + y
}

async function  main() {
    console.log('1')
    add(1,2)
    console.log('3')
    addAsync(1,2)
    console.log('5')
}


async function  mainAsync() {
    console.log('1')
    add(1,2)
    console.log('3')
    await addAsync(1,2)
    console.log('5')
}

main()
mainAsync()

```
<img width="305" alt="Screen Shot 2020-04-01 at 5 19 41 PM" src="https://user-images.githubusercontent.com/10264973/78198355-30df7780-743d-11ea-9134-5dc0a45ef549.png">

Reply:
ha, played around with `await` some more and I think I see what you mean:
```
> f1 = async() => { await(sleep(100).then(() => console.log('f1'))) }
[AsyncFunction: f1]
> f2 = async() => { await(f1()); console.log('f2') }
[AsyncFunction: f2]
> f3 = async() => { f1(); console.log('f3') }
[AsyncFunction: f3]
> f2()
f1
f2
> f3()
f3
f1

```
I'll add it here.

Reply:
Ya it’s got a couple gotchas @shashiranjan84 did you find any good resources on js event loop?

Reply:
https://snyk.io/blog/nodejs-how-even-quick-async-functions-can-block-the-event-loop-starve-io/ -> very well explained

Reply:
added this to our notion stack page: https://www.notion.so/transcend/Node-c050db956d9c4cc98796a9a779be2e04